File: app/src/main/java/com/mancode/financetracker/database/entity/CurrencyEntity.java
package com.mancode.financetracker.database.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

/**
 * Created by Manveru on 23.01.2018.
 */

@Entity(tableName = CurrencyEntity.TABLE_NAME)
public class CurrencyEntity {

    public static final String TABLE_NAME = "currencies";

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    private int id;

    @ColumnInfo(name = "currency_symbol")
    private String currencySymbol;

    @ColumnInfo(name = "currency_exchange_rate")
    private double exchangeRate;

    @Ignore
    public CurrencyEntity() {

    }

    public CurrencyEntity(int id, String currencySymbol, double exchangeRate) {
        initFromValues(id, currencySymbol, exchangeRate);
    }

    public void initFromValues(int id, String currencySymbol, double exchangeRate) {
        this.id = id;
        this.currencySymbol = currencySymbol;
        this.exchangeRate = exchangeRate;

    }

    public int getId() {
        return id;
    }

    public String getCurrencySymbol() {
        return currencySymbol;
    }

    public double getExchangeRate() {
        return exchangeRate;
    }
}

File: app/src/main/java/com/mancode/financetracker/database/entity/CurrencyEntity.kt
package com.mancode.financetracker.database.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.mancode.financetracker.database.converter.DateConverter
import org.threeten.bp.LocalDate

/**
 * Created by Manveru on 23.01.2018.
 */

@Entity(tableName = CurrencyEntity.TABLE_NAME)
@TypeConverters(DateConverter::class)
data class CurrencyEntity (

    @PrimaryKey @ColumnInfo(name = "currency_symbol") val currencySymbol: String,
    @ColumnInfo(name = "currency_exchange_rate") var exchangeRate: Double,
    @ColumnInfo(name = "currency_rate_date") var rateDate: LocalDate?
) {

    companion object {
        const val TABLE_NAME = "currencies"
    }
}
